examples/cholesky/dask_gpu_cholesky.py

```diff
@@ -0,0 +1,158 @@
+import os
+import mkl
+import argparse
+import operator
+from functools import partial
+
+parser = argparse.ArgumentParser()
+parser.add_argument('-workers', type=int, default=1)
+parser.add_argument('-t', type=int, default=1)
+parser.add_argument('-process', type=int, default=0)
+args = parser.parse_args()
+
+is_process = True if args.process else False
+
+mkl.set_num_threads(args.t)
+os.environ["NUMEXPR_NUM_THREADS"] = str(args.t)
+os.environ["OMP_NUM_THREADS"] = str(args.t)
+os.environ["OPENBLAS_NUM_THREADS"] = str(args.t)
+os.environ["VECLIB_MAXIMUM_THREADS"] = str(args.t)
+
+import pprint
+from dask_cuda import LocalCUDACluster
+from dask.distributed import Client
+from dask.array.core import Array
+from dask.array.utils import array_safe, meta_from_array, solve_triangular_safe
+from dask.base import tokenize
+from dask.highlevelgraph import HighLevelGraph
+from dask.diagnostics import ProgressBar
+
+import dask.array
+import dask
+import numpy as np
+import cupy as cp
+from cupy.cuda import cublas
+from cupy.cuda import device
+from scipy.linalg.blas import dtrsm
+
+import time
+
+def potrf(a):
+    a = cp.linalg.cholesky(a)
+    return a
+
+
+def cupy_trsm_wrapper(a, b):
+    cublas_handle = device.get_cublas_handle()
+    trsm = cublas.dtrsm
+    uplo = cublas.CUBLAS_FILL_MODE_LOWER
+
+    a = cp.asarray(a, dtype=np.float64, order='F')
+    b = cp.asarray(b, dtype=np.float64, order='F')
+
+    trans = cublas.CUBLAS_OP_T
+    side = cublas.CUBLAS_SIDE_RIGHT
+
+    diag = cublas.CUBLAS_DIAG_NON_UNIT
+    m, n = (b.side, 1) if b.ndim == 1 else b.shape
+    trsm(cublas_handle, side, uplo, trans, diag, m, n, 1.0, a.data.ptr, m, b.data.ptr, m)
+    return b
+
+
+def solve(a, b):
+    b = cupy_trsm_wrapper(a, b)
+    return b
+
+def handle_trsm(a, b):
+    b = solve(a, b)
+    return b
+
+
+def gemm(a, b):
+    return a@b.T
+
+
+def syrk(a):
+    return a@a.T
+
+
+def zerofy(a):
+    return partial(cp.zeros_like, shape=(a.chunks[0][i], a.chunks[1][j]))
+
+
+def blocked_cholesky(a):
+    num_blocks = len(a.chunks[0])
+    token = tokenize(a)
+    name = "cholesky-" + token
+    name_lt_dot = "cholesky-lt-dot-" + token
+
+    dsk = {}
+    print("num_blocks:", num_blocks)
+    for i in range(num_blocks):
+        a_ii = (a.name, i, i)
+        if i > 0:
+            prevs = []
+            for k in range(i):
+                prev = name_lt_dot, i, k, i, k
+                dsk[prev] = (syrk, (name, i, k))
+                prevs.append(prev)
+            a_ii = (operator.sub, a_ii, (sum, prevs))
```

Comment:
I think this sum might cause a CPU copy (mostly leaving this as a note to check this) before I forget.